pkg/cli/alpha/config-gen/cmd.go

```diff
@@ -252,40 +251,48 @@ EOF
 kustomize build --enable-alpha-plugins .
 `)
 
-	dir, dirErr := getPluginDir()
-	pluginFile := filepath.Join(dir, "KubebuilderConfigGen")
-
 	// command for installing the plugin
 	install := &cobra.Command{
 		Use:   "install-as-plugin",
 		Short: "Install config-gen as a kustomize plugin",
-		Long: strings.TrimSpace(fmt.Sprintf(`
-Write a script to %s for kustomize to locate as a plugin.
-`, pluginFile)),
-		Example: strings.TrimSpace(`
+		Long: fmt.Sprintf(`Write a script to %s for kustomize to locate as a plugin.
+This path will be written to $XDG_CONFIG_HOME if set, otherwise $HOME.
+`, pluginScriptPath),
+		Example: `
 kubebuilder alpha config-gen install-as-plugin
-`),
+`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if dirErr != nil {
-				return dirErr
+			hd, err := getPluginHomeDir()
+			if err != nil {
+				log.Fatal(err)
 			}
-			fmt.Fprintf(cmd.OutOrStdout(), "writing kustomize plugin file at %s\n", pluginFile)
+			fullScriptPath := filepath.Join(hd, pluginScriptPath)
+
+			fmt.Fprintf(cmd.OutOrStdout(), "writing kustomize plugin file at %s\n", fullScriptPath)
+
+			dir, _ := filepath.Split(fullScriptPath)
 			err = os.MkdirAll(dir, 0700)
 			if err != nil {
 				return err
 			}
 
-			return ioutil.WriteFile(pluginFile, []byte(`#!/bin/bash 
+			return ioutil.WriteFile(fullScriptPath, []byte(`#!/bin/bash
 KUSTOMIZE_FUNCTION=true kubebuilder alpha config-gen
-`), 0500)
+`), 0700)
```

Comment:
These perms let the command overwrite this file. Otherwise rerunning the command will fail. @pwittrock are you ok with this for now?

Reply:
We can ignore here the gosec here.

Reply:
Removed this perm change. The command doesn't attempt to overwrite the file, just prints that it exists.